src/server/array_func.c

```diff
@@ -137,6 +137,257 @@ static enum job_atr attrs_to_copy[] = {
 	JOB_ATR_LAST /* This MUST be LAST	*/
 };
 
+
+/**
+ * @brief
+ * 			init_range - Initialize the sub-job indices
+ *
+ * @param[in]	rarr - pointer to the range_arr_t 
+ * @param[in]	start - start of the subjob indices
+ * @param[in]	end - end of the subjob indices
+ * @param[in]	step - step factor in subjob indices
+ *
+ * @return      Int
+ * @retval	 0  - Intialization success
+ * @retval	 1  - failure 
+ */
+
+int init_range(range_arr_t *rarr, int start, int end, int step)
+{
+	rarr->count = 1;
+	rarr->step= step;
+	rarr->ranges = malloc(sizeof(range_t));
+
+	rarr->ranges[0].start = start;
+	rarr->ranges[0].end = end;
+	return 0;
+}
+
+/**
+ * @brief
+ * 			add_value - Adding a value in the range structures
+ * 						handles adding value in mid of the ranges.
+ *
+ * @param[in]	rarr - pointer to the range_arr_t 
+ * @param[in]	val - Value to add
+ *
+ * @return      Int
+ * @retval	 0  - Intialization success
+ * @retval	 1  - failure 
+ */
+
+int
+add_value(range_arr_t *rarr, int val)
+{
+	int i = 0,j = 0;
+	range_t *new;
+	
+	if (rarr->ranges == NULL) {
+		init_range(rarr, val, val, rarr->step);
+		return 0;
+	}
+
+	/*check the extreme left corner case */
+	if ( val < rarr->ranges[i].start) {
+		rarr->count++;
+		new = realloc(rarr->ranges, sizeof(range_t) * rarr->count);
+		if (new)
+			rarr->ranges = new;
+		else
+			return 1;
+		for (j = rarr->count - 1; j > i; j--) { /* move the pointers right */
```

Comment:
however this loop and above realloc will be less in number but using sched's range struct we can get rid of this loop and above realloc